book/src/qobject/qobject_struct.md

```diff
@@ -5,43 +5,158 @@ SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
-  * Invokable methods that are exposed to Qt
-  * Private methods and fields for struct to use (eg this is useful for storing the channels for [threading](../concepts/threading.md))
-  * Mutate C++ state with [`CppObj`](./cpp_object.md)
-  * Use [CxxQtThread<T>](./cxxqtthread.md) for multi threaded workflows
+Fields within the `#[cxx_qt::qobject]` marked struct can be tagged with `#[qproperty]` to be exposed as [`Q_PROPERTY`s](https://doc.qt.io/qt-6/properties.html) on the generated QObject:
 
 ```rust,ignore,noplayground
-{{#include ../../../examples/qml_features/rust/src/rust_obj_invokables.rs:book_macro_code}}
+{{#include ../../../examples/qml_features/rust/src/properties.rs:book_properties_struct}}
 ```
 
-## Invokables
+Any type that CXX supports may be marked as a `#[qproperty]`.
+See the [Types page](../concepts/types.md) for a list of supported types.
 
-A `impl qobject::T` is used to define invokables, the `impl qobject::T` defines that the methods are implemented onto the C++ QObject `T`.
-Therefore they have access to both C++ and Rust methods. Also CXX-Qt adds wrapper code around your invokables to automatically perform any conversion between the [C++ and Rust types](../concepts/types.md).
+For every `#[qproperty]`, CXX-Qt will generate setters and getters, as well as a "changed" signal.
 
-To mark a method as invokable simply add the `#[qinvokable]` attribute to the Rust method. This then causes `Q_INVOKABLE` to be set on the C++ definition of the method, allowing QML to call the invokable.
+On the C++ side:
+  * setter: `set<Property>`
+  * getter: `get<Property>`
+  * changed: `<Property>Changed`
 
-Note to access properties on the C++ object use [Cpp Object](./cpp_object.md).
+On the Rust side:
+  * setter: `set_<Property>`
+  * getter: `<Property>`
+  * changed: `<Property>_changed`
 
-## Properties
+where `<Property>` is the name of the property.
+
+These setters and getters assure that the changed signal is emitted every time the property is edited.
+
+Any field that's not marked as `#[qproperty]` won't be accessible from C++, but it will be accessible from Rust.
+See the [Private fields section](#private-methods-and-fields)
+
+### `cxx_type`
+
+You can change the C++ type that your property uses by adding the `cxx_type` attribute to the `#[qproperty]` macro.
+
+This is especially useful if your property type is an opaque C++ type and can't be owned directly by Rust.
+Then it might be necessary to wrap it in a [cxx::UniquePtr](https://docs.rs/cxx/latest/cxx/struct.UniquePtr.html).
+
+However, you may still want your `Q_PROPERTY` to use the opaque type directly.
+This can easily be achieved by using `cxx_type`.
+
+``` rust,ignore,noplayground
+#[cxx_qt::qobject]
+struct MyStruct {
+  #[qproperty(cxx_type="OpaqueExampleType")]
+  my_property: cxx::UniquePtr<OpaqueExampleType>
+}
+```
+In this case, CXX-Qt will automatically convert any references to the `std::unique_ptr<OpaqueExampleType>` to references to `OpaqueExampleType`.
```

Comment:
```suggestion
In this case, CXX-Qt will automatically conver the `std::unique_ptr<OpaqueExampleType>` to `OpaqueExampleType`.
```

This could be confused with Rust and/or C++ references (which would be odd on a unique_ptr).

Reply:
done